sm/download/src/download.rs

```diff
@@ -25,15 +34,25 @@ impl Downloader {
 
         let target_filename = PathBuf::new().join(target_dir_path).join(filename);
 
+        let config_location = TEdgeConfigLocation::from_default_system_location();
+        let config_repository = tedge_config::TEdgeConfigRepository::new(config_location.clone());
+        let tedge_config = config_repository.load().unwrap();
+
+        let buffer_size: Buffer = tedge_config
+            .query(SoftwareDownloadBufferSizeDefaultSetting)
+            .unwrap();
+
         Self {
             target_filename,
             download_target,
+            reserved_buffer_size: buffer_size.apply_to(fs2::total_space("/tmp").unwrap()) as i64,
```

Comment:
In crate `nix` which you are already using anyway for `fallocate` there is `stavfs` function to extract that info as well. I would say it is not worth importing the whole crate for just this one function use.

[statvfs](https://docs.rs/nix/0.23.0/nix/sys/statvfs/index.html)

Reply:
The nix has an unsafe code. But fs2 is not, just thought of using it.

Reply:
I'm afraid both crates use the same method which is ffi system call to `stavfs` which is C code. fs2 is nicely hiding it within a module though.